### ISSUE TITLE
setModuleDefaults overwriting loaded settings

### ISSUE THREAD

**Issue by msamblanet** [lorenwest/node-config#145]
With more use of 1.2.3 after #142 was fixed, I found another issue.  It seems like in some cases that setModuleDefaults is overwriting my configuration file values.  The sample below demonstrates the issue.

I have not had a lot of time to look in the node-config code yet but best guess is at 387, it is doing a deepCopy of the new defaults over the configuration properties - this does not feel right since deep copy overwrites conflicting properties...but this was just a quick scan - I could be wrong on this.

// config/local.js:
//    module.exports = {
//        a: { foo: 1 }
//    };
var config = require("config");
console.log(JSON.stringify(config.util.getConfigSources(), 0, 2)); // To confirm right config data
config.util.setModuleDefaults("a", {
    foo: 0
});
var cfg = config.get("a");
console.log(JSON.stringify(cfg, 0, 2));
//   With 1.2.3 output is foo:0 not foo:1 (incorrect)
//   With 1.1.0 output is foo:1 as expected
